apps/meteor/app/livechat/server/lib/contacts/registerGuestData.ts

```diff
@@ -35,5 +29,9 @@ export async function registerGuestData(
 		return;
 	}
 
-	await ContactMerger.mergeFieldsIntoContact(fields, contact, contact.unknown ? 'overwrite' : 'conflict');
+	// If a visitor was updated who already had contacts, load up the contacts and update that information as well
+	const contacts = await LivechatContacts.findAllByVisitorId(visitor._id).toArray();
+	for await (const contact of contacts) {
+		await ContactMerger.mergeFieldsIntoContact(fields, contact, contact.unknown ? 'overwrite' : 'conflict');
```

Comment:
Will this fix [SCI-136](https://rocketchat.atlassian.net/browse/SCI-136)?

Reply:
Probably not by itself, as this was already being done before.